test/ng/directive/inputSpec.js

```diff
@@ -2819,6 +2819,313 @@ describe('input', function() {
+      // allow to set a non-number value and will set the value of the input to 50 even when you
+      // change it directly on the element.
+      // Other browsers fall back to text inputs, where setting a model value of 50 does not make
+      // sense if the input value is a string. These browser will mark the input as invalid instead.
+      it('should parse non-number values to 50', function() {
+        var inputElm = helper.compileInput('<input type="range" ng-model="age" />');
+
+        scope.$apply('age = 10');
+        expect(inputElm.val()).toBe('10');
+
+        helper.changeInputValueTo('');
+        expect(scope.age).toBe(50);
+        expect(inputElm).toBeValid();
+      });
+    }
+
+    it('should only invalidate the model if suffering from bad input when the data is parsed', function() {
+      scope.age = 60;
+
+      var inputElm = helper.compileInput('<input type="range" ng-model="age" />', {
+        valid: false,
+        badInput: true
+      });
+
+      expect(inputElm).toBeValid();
+
+      helper.changeInputValueTo('this-will-fail-because-of-the-badInput-flag');
+
+      expect(scope.age).toBeUndefined();
+      expect(inputElm).toBeInvalid();
+    });
+
+
+    it('should validate with undefined viewValue when $validate() called', function() {
+      var inputElm = helper.compileInput('<input type="range" name="alias" ng-model="value" />');
+
+      scope.form.alias.$validate();
+
+      expect(inputElm).toBeValid();
+      expect(scope.form.alias.$error.number).toBeUndefined();
+    });
+
+
+    it('should throw if the model value is not a number', function() {
+      expect(function() {
+        scope.value = 'one';
+        var inputElm = helper.compileInput('<input type="range" ng-model="value" />');
+      }).toThrowMinErr('ngModel', 'numfmt', "Expected `one` to be a number");
+    });
+
+
+    describe('min', function() {
+
+      if (msie !== 9) {
+        // Browsers that implement range will never allow you to set the value < min values
+        it('should validate', function() {
+          var inputElm = helper.compileInput('<input type="range" ng-model="value" name="alias" min="10" />');
+
+          helper.changeInputValueTo('5');
+          expect(inputElm).toBeValid();
+          expect(scope.value).toBe(10);
+          expect(scope.form.alias.$error.min).toBeFalsy();
+
+          helper.changeInputValueTo('100');
+          expect(inputElm).toBeValid();
+          expect(scope.value).toBe(100);
+          expect(scope.form.alias.$error.min).toBeFalsy();
+        });
+      }
+
+      it('should validate if "range" is not implemented', function() {
+        var inputElm = helper.compileInput('<input type="range" ng-model="value" name="alias" min="10" />');
+
+        // To test how browsers that don't implement range behave, we have to change the type, so
+        // that special layout and validation rules don't apply
+        inputElm[0].setAttribute('type', 'text');
```

Comment:
I think this might introduce some funny behavior that is different from the actuall behavior (not to mention you said IE8 will throw - not that we care :stuck_out_tongue: ).
I suggest we check if the type is not supported (e.g. `inputElm[0].type === 'text'`) and only then continue with the test.